pep-0681.rst

```diff
@@ -0,0 +1,709 @@
+fields. This is not currently supported in this spec, but it could be
+added in the future if there was sufficient demand.
+
+Automatic field name aliasing
+-----------------------------
+The attrs library performs automatic aliasing of field names that
+start with a single underscore. This proposal omits this behavior.
+
+Alternate field ordering algorithms
+-----------------------------------
+The attrs library currently supports two approaches to ordering the
+fields within a class:
+
+* Dataclass order: The same ordering used by dataclasses. This is the
+  default behavior of the older APIs (ex. attr.s).
+* Method Resolution Order (MRO): This is the default behavior of the
+  newer APIs (ex. define, mutable, frozen). Older APIs (ex. attr.s)
+  can opt into this behavior by specifying ``collect_by_mro=True``.
+
+The resulting field orderings can differ in certain diamond-shaped
+multiple inheritance scenarios.
+
+This proposal does not support any field ordering other than that used
+by dataclass.
+
+Fields redeclared in subclasses
+-------------------------------
+The attrs library differs from stdlib dataclasses in how it
+handles inherited fields that are redeclared in subclasses. The
+dataclass specification preserves the original order, but attrs
+defines a new order based on subclasses. Users of attrs who rely on
+this ordering will not see the expected order of parameters in the
+synthesized ``__init__`` method.
+
+Django primary and foreign keys
+-------------------------------
+Django applies additional logic for primary keys and
+foreign keys. For example, it automatically adds an "id" field (and
+``__init__`` parameter) if there is no field designated as a primary
+key. This additional logic is not accommodated with this proposal, so
+users of Django would need to explicitly declare the id field.
+
+This limitation may make it impractical to use the
+``dataclass_transform`` mechanism with Django.
+
+Open Issues
+===========
+
+``converter`` field descriptor parameter
+----------------------------------------
+The attrs library supports a ``converter`` field descriptor parameter,
+which is a callable that is called by the generated
+``__init__`` method to convert the supplied value to some other
+desired value. This is tricky to support since the parameter type in
+the synthesized __init__ method needs to accept uncovered values, but
+the resulting field is typed according to the output of the converter.
+
+There may be no good way to support this because there's not enough
+information to derive the type of the input parameter. We currently
+have two ideas:
+
+1. Add support for a ``converter`` field descriptor parameter but then
+   use the Any type for the corresponding parameter in the  __init__
+   method.
+
+2. Say that converters are unsupported and recommend that attrs users
+   avoid them.
+
+Some aspects of this issue are discussed on the pyright GitHub site.
+[#converters]_
```

Comment:
```suggestion
Some aspects of this issue are detailed in a
`Pyright discussion <#converters_>`_
```

Example of how this would look and fix caps, wording, etc.